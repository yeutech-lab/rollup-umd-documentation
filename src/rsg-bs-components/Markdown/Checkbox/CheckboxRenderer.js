import React from 'react';
import Input from 'bootstrap-styled/lib/Input';

export function CheckboxRenderer({ ...rest }) {
  return <Input {...rest} type="checkbox" />;
}

export default CheckboxRenderer;
