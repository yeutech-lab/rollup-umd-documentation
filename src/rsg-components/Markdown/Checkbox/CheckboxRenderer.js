import React from 'react';
import PropTypes from 'prop-types';

import Styled from 'rsg-components/Styled';

const styles = () => ({
	input: {
		isolate: false,
		display: 'inline-block',
		verticalAlign: 'middle',
	},
});

export function CheckboxRenderer({ classes, ...rest }) {
	return <input {...rest} type="checkbox" className={classes.input} />;
}
CheckboxRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
};

export default Styled(styles)(CheckboxRenderer);
