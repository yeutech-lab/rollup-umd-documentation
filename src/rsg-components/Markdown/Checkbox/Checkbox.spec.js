import React from 'react';

import Checkbox from './index';

describe('Markdown Checkbox', () => {
	it('should render a checkbox input', () => {
		const actual = render(<Checkbox />);

		expect(actual).toMatchSnapshot();
	});
});
