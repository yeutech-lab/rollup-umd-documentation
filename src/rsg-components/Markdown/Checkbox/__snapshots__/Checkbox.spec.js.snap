// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown Checkbox should render a checkbox input 1`] = `
<input
  class="rsg--input-0"
  type="checkbox"
/>
`;
